src/logic/contracts/getSafeCreationGasEstimate.tsx

```diff
@@ -44,3 +45,5 @@ export const getEstimateSafeCreationGas = async ({
     gasLimit: gasEstimation,
   }
 }
+
+export const getSafeCreationGasEstimate = throttle(estimateSafeCreationGas, 10000)
```

Comment:
Please put 10 seconds also in a constant.

Reply:
Is a safe creation the only tx where we get the gas price? What about regular txns?

Reply:
Another concern is that the function we're trying to throttle is async. Does `_.throttle` play nicely with them?
Alternatively there are packages like [p-throttle](https://github.com/sindresorhus/p-throttle).